drivers/pinctrl/pinctrl_sam.c

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2022, Gerson Fernando Budke <nandojve@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/pinctrl.h>
+#include <soc_gpio.h>
+
+/** Utility macro that expands to the GPIO port address if it exists */
+#define SAM_PORT_ADDR_OR_NONE(nodelabel)					\
+	COND_CODE_1(DT_NODE_EXISTS(DT_NODELABEL(nodelabel)),			\
+		   (DT_REG_ADDR(DT_NODELABEL(nodelabel)),), ())
```

Comment:
Just FYI, you can use `IF_ENABLED(DT_NODE_EXISTS(DT_NODELABEL(nodelabel)), (DT_REG_ADDR(...)))` instead if you want to save some typing.

Reply:
Yeah, it makes cleaner, tks